google/cloud/spanner/keys_test.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/keys.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace {
+
+TEST(KeyRangeTest, ConstructorBoundModeUnspecified) {
+  std::string start_value("key0");
+  std::string end_value("key1");
+  KeyRange<Row<std::string>> closed_range =
+      KeyRange<Row<std::string>>(MakeRow(start_value), MakeRow(end_value));
```

Comment:
I think we want a `MakeKeyRange` helper that deduces the RowType. Then this will become:
```cc
auto closed_range = MakeKeyRange(MakeRow(a, b, c), MakeRow(x, y, z));
```